builtin/rebase.c

```diff
@@ -166,5 +186,45 @@ int cmd_rebase(int argc, const char **argv, const char *prefix)
 		break;
 	}
 
-	return !!run_specific_rebase(&options);
+	options.upstream_name = argv[1];
+	if (!strcmp(options.upstream_name, "-"))
+		options.upstream_name = "@{-1}";
+
+	options.upstream = peel_committish(options.upstream_name);
+	if (!options.upstream)
+		die(_("invalid upstream '%s'"), options.upstream_name);
+
+	/* Make sure the branch to rebase onto is valid. */
+	options.onto_name = options.onto ?
+		options.onto : options.upstream_name;
+	if (strstr(options.onto_name, "...")) {
+		die("TODO");
+	} else {
+		options.onto = peel_committish(options.onto_name);
+		if (!options.onto)
+			die(_("Does not point to a valid commit '%s'"),
+				options.onto_name);
+	}
+
+	/* Do not need to switch branches, we are already on it. */
```

Comment:
After doing this, I think we should skip these lines for now: https://github.com/prertik/git/blob/4ba1302de224cf719a7492d5970aca511dd13024/git-legacy-rebase.sh#L560-L613 (because they are not really essential to run a `rebase --am`).

So that leaves https://github.com/prertik/git/blob/4ba1302de224cf719a7492d5970aca511dd13024/git-legacy-rebase.sh#L615-L620 and https://github.com/prertik/git/blob/4ba1302de224cf719a7492d5970aca511dd13024/git-legacy-rebase.sh#L632-L639 to be converted and we can perform our first real builtin rebase!